blas/testblas/zhemm.go

```diff
@@ -46,81 +47,90 @@ func zhemmTest(t *testing.T, impl Zhemmer, side blas.Side, uplo blas.Uplo, m, n
 			for _, ldc := range []int{max(1, n), n + 4} {
 				for _, alpha := range []complex128{0, 1, complex(0.7, -0.9)} {
 					for _, beta := range []complex128{0, 1, complex(1.3, -1.1)} {
-						// Allocate the matrix A and fill it with random numbers.
-						a := make([]complex128, nA*lda)
-						for i := range a {
-							a[i] = rndComplex128(rnd)
-						}
-						// Create a copy of A for checking that
-						// Zhemm does not modify its triangle
-						// opposite to uplo.
-						aCopy := make([]complex128, len(a))
-						copy(aCopy, a)
-						// Create a copy of A expanded into a
-						// full hermitian matrix for computing
-						// the expected result using zmm.
-						aHem := make([]complex128, len(a))
-						copy(aHem, a)
-						if uplo == blas.Upper {
-							for i := 0; i < nA; i++ {
-								aHem[i*lda+i] = complex(real(aHem[i*lda+i]), 0)
-								for j := i + 1; j < nA; j++ {
-									aHem[j*lda+i] = cmplx.Conj(aHem[i*lda+j])
+						for _, nanC := range []bool{false, true} {
+							if nanC && beta != 0 {
```

Comment:
Can you please add a comment explaining why we are not interested in this test case? Also at the other places.

Reply:
Done